mobile/src/main/java/ch/epfl/sdp/mobile/ui/ar/ArScreen.kt

```diff
@@ -32,42 +60,107 @@ fun ArScreen(modifier: Modifier = Modifier) {
-    val instance =
-        node.loadModel(
-            context = context,
-            glbFileLocation = glbPath,
-            autoScale = true,
-        )
-    val boundingBox = instance?.filamentAsset?.boundingBox
-    val halfExtent = boundingBox?.halfExtent?.maxOrNull()!!
-    // TODO : Determine the scale
-    node.scale(0.5f / halfExtent)
+    node.loadModel(
+        context = context,
+        glbFileLocation = glbPath,
+        autoScale = true,
+    )
     return node
   }
 
-  /**
-   * Add the given [ArModelNode] to the scene
-   *
-   * @param modelNode the model that we want to add
-   * @param view Where the model will be added
-   */
-  fun addNode(modelNode: ArModelNode?, view: ArSceneView) {
-    val node = modelNode ?: return
-    if (node !in view.children) {
-      view.addChild(node)
-    }
+  // Keep the screen only for this composable
+  DisposableEffect(Unit) {
+    view.keepScreenOn = true
+    onDispose { view.keepScreenOn = false }
   }
 
+  // Load 3d Model and initialize the [ArBoard]
   LaunchedEffect(Unit) {
+    // TODO : Maybe create a enum with all the pieces that contain the path
     pawn1 = loadModel("models/pawn.glb", nodePlacementMode)
-    pawn2 = loadModel("models/pawn.glb", nodePlacementMode, Position(x = 1f))
+    board = loadModel(BoardPath, nodePlacementMode)
+
+    val boardBoundingBox = board!!.modelInstance?.filamentAsset?.boundingBox!!
+
+    // get height (on y axe) of the board
+    // Double the value to get the total height of the box
+    val boardYOffset = 2 * boardBoundingBox.halfExtent[1]
+    val boardHalfSize = boardBoundingBox.halfExtent[0]
+
+    arBoard = ArBoard(BoardBorderSize, boardYOffset, boardHalfSize)
+
+    // DEBUG
+    if (DisplayAxes) {
+      white = loadModel("models/white.glb", nodePlacementMode)
+      red = loadModel("models/red.glb", nodePlacementMode)
+      blue = loadModel("models/blue.glb", nodePlacementMode)
+      green = loadModel("models/green.glb", nodePlacementMode)
+    }
   }
 
   AndroidView(
       factory = { ArSceneView(it) },
-      modifier = modifier,
+      modifier = modifier.testTag("AR"),
       update = { view ->
-        addNode(pawn1, view)
-        addNode(pawn2, view)
+        if (DisplayAxes) {
+          white?.let {
+            it.scale(2f)
+            board?.addChild(it)
+            it.placementPosition = Position(y = arBoard!!.boardHeight)
+          }
+          red?.let {
+            it.scale(2f)
+            board?.addChild(it)
+            it.placementPosition = Position(x = 3f, y = arBoard!!.boardHeight)
+          }
+          blue?.let {
+            it.scale(2f)
+            board?.addChild(it)
+            it.placementPosition = Position(z = 3f, y = arBoard!!.boardHeight)
+          }
+          green?.let {
+            it.scale(2f)
+            board?.addChild(it)
+            it.placementPosition = Position(y = 3f + arBoard!!.boardHeight)
+          }
+        }
```

Comment:
Question : is there a scenario where `white`, `red`, `blue` or `green` might be non-null but `DisplayAxes` is false ?

Reply:
Non because we only load the model if `DisplayAxes` is true (line 93-98).